gate-iap/src/main/java/com/netflix/spinnaker/gate/security/iap/IapAuthenticationFilter.java

```diff
@@ -144,27 +148,31 @@ private boolean isServiceAccount(String email) {
   private User verifyJWTAndGetUser(SignedJWT jwt) throws Exception {
     JWSHeader jwsHeader = jwt.getHeader();
 
-    Preconditions.checkNotNull(jwsHeader.getAlgorithm());
-    Preconditions.checkNotNull(jwsHeader.getKeyID());
+    Preconditions.checkNotNull(jwsHeader.getAlgorithm(), "JWT header is missing algorithm (alg)");
+    Preconditions.checkNotNull(jwsHeader.getKeyID(), "JWT header is missing key ID (kid)");
 
     JWTClaimsSet claims = jwt.getJWTClaimsSet();
 
-    Preconditions.checkArgument(claims.getAudience().contains(configProperties.audience));
-    Preconditions.checkArgument(claims.getIssuer().contains(configProperties.issuerId));
+    Preconditions.checkArgument(claims.getAudience().contains(configProperties.getAudience()),
+      String.format("JWT payload audience claim (aud) must contain: %s.", configProperties.getAudience()));
+    Preconditions.checkArgument(claims.getIssuer().equals(configProperties.getIssuerId()),
+      String.format("JWT payload issuer claim (iss) must be: %s", configProperties.getIssuerId()));
 
-    Date currentTime = new Date();
-    Preconditions.checkArgument(claims.getIssueTime().before(currentTime));
-    Preconditions.checkArgument(claims.getExpirationTime().after(currentTime));
+    Date currentTime = Date.from(Instant.now(CLOCK));
+    Preconditions.checkArgument(claims.getIssueTime().before(new Date(currentTime.getTime() + configProperties.getIssuedAtTimeAllowedSkew())),
```

Comment:
This is pre-existing to this PR, but should we be checking the `notBeforeTime` instead of the `issueTime` here?  It looks like the standards state that use of `issueTime` is optional but that implementers must verify the `notBeforeTime`.

https://tools.ietf.org/html/rfc7519#section-4.1.5

Reply:
The IAP docs do ask us to verify the `issuedAtTime` though, I assume since they issue the jwt token they chose to use this instead of `nbt`
https://cloud.google.com/iap/docs/signed-headers-howto

Reply:
Interesting---makes sense then!